test/qa/test_combinations.py

```diff
@@ -112,50 +41,26 @@ def test_reconnect_matrix_obfuscated(
         target_obfuscated,
 ):
     lib.set_technology_and_protocol(source_tech, source_proto, source_obfuscated)
-
-    output = sh.nordvpn.connect(_tty_out=False)
-    print(output)
-    assert lib.is_connect_successful(output)
-    assert network.is_connected()
+    connect_base_test(ipv6 = False)
 
     lib.set_technology_and_protocol(target_tech, target_proto, target_obfuscated)
+    connect_base_test(ipv6 = False)
 
-    output = sh.nordvpn.connect(_tty_out=False)
-    print(output)
-    assert lib.is_connect_successful(output)
-    assert network.is_connected()
-
-    output = sh.nordvpn.disconnect()
-    print(output)
-    assert lib.is_disconnect_successful(output)
-    assert network.is_disconnected()
+    disconnect_base_test()
 
 
+@pytest.mark.parametrize(("tech", "proto", "obfuscated"), lib.TECHNOLOGIES)
 @pytest.mark.parametrize(("country", "city"), list(zip(lib.COUNTRIES, lib.CITIES, strict=False)))
 @pytest.mark.flaky(reruns=2, reruns_delay=90)
 @timeout_decorator.timeout(40)
-def test_connect_country_and_city(country, city):
-    lib.set_technology_and_protocol("nordlynx", "", "")
-
-    output = sh.nordvpn.connect(country, _tty_out=False)
-    print(output)
-    assert lib.is_connect_successful(output)
-    assert network.is_connected()
-
-    output = sh.nordvpn.connect(city, _tty_out=False)
-    print(output)
-    assert lib.is_connect_successful(output)
-    assert network.is_connected()
+def test_connect_country_and_city(tech, proto, obfuscated, country, city):
+    lib.set_technology_and_protocol(tech, proto, obfuscated)
 
-    output = sh.nordvpn.connect(country, city, _tty_out=False)
-    print(output)
-    assert lib.is_connect_successful(output)
-    assert network.is_connected()
+    connect_base_test(country, ipv6 = False)
```

Comment:
shouldn't there be a `disconnect` or is tested server switching?

Reply:
Reconnect combinations are being tested, so `disconnect` is not needed in this case